relay-server/src/services/processor/span/processing.rs

```diff
@@ -98,6 +102,19 @@ pub fn process(
             _ => return ItemAction::Keep,
         };
 
+        if let Some(span) = annotated_span.value() {
```

Comment:
After some different implementations, I settled on just putting the filtering here, even though I don't know if it should be put into the `process` function. The only thing I know is that separating into two functions would require quite a bit of refactoring.